biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -75,21 +80,90 @@
 	private String				queryUrl;
 	private ScheduledFuture<?>	pomPoller;
 
-	public synchronized void init() {
+	private String				status;
+
+	private PromiseFactory		promiseFactory		= new PromiseFactory(Executors.newSingleThreadExecutor());
```

Comment:
You should get this one from the registry (setRegistry), or `Processor.getPromiseFactory()`

Reply:
good to know, I will use this factory instead